src/components/EntryForm/EntryForm.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import { useEntries } from '../../context/EntriesContext';
+import { useUser } from '../../context/UserContext';
+import './EntryForm.css';
+
+export default function EntryForm() {
+  const { user, setUser } = useUser();
+  const { entries, setEntries } = useEntries();
+
+  const [userInput, setUserInput] = useState('');
+  const [message, setMessage] = useState('');
+
+  const formSubmit = (e) => {
+    e.preventDefault();
+    setUser(userInput);
+    const id = entries.length;
+    setEntries([...entries, { user: userInput, message, id }]);
```

Comment:
might be a good idea to spread in prevState here instead of entries to make sure its correctly updated